### PR TITLE
Clean up dockershim in tests

### PR DESCRIPTION
**Pull request by hakman**
#### What type of PR is this?
/kind cleanup

#### What this PR does / why we need it:
As part of the removal of dockershim, we should also clean up the tests.

#### Which issue(s) this PR fixes:

Refs #106893

#### Special notes for your reviewer:
n/a

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

File: test/e2e/framework/test_context.go [hunk 1]
```diff
@@ -486,6 +486,17 @@ func AfterReadingAllFlags(t *TestContextType) {
 		TestContext.Provider = "skeleton"
 	}
 
+	// TODO: Fix tests scripts that set CONTAINER_RUNTIME="containerd"
```
[the remark this answers is not in the document]
Comment (hakman) [reply test/e2e/framework/test_context.go:489]: Yes, that would be prefereable.

**Review by SergeyKanzhelev — APPROVED** [review]
/triage accepted
/lgtm
/priority important-soon

**Comment by dims** [comment]
/assign @johnbelamaric 

@spiffxp @johnbelamaric : very easy to review ... just the conformance text file update! rest of the stuff has already been lgtmed

**Pull request merged**